pkgs/os-specific/linux/lenovo-legion/default.nix

```
@@ -0,0 +1,42 @@
+{ lib, fetchurl, stdenv, kernel, bash, pkgs }:
+
+stdenv.mkDerivation {
+  pname = "lenovo-legion";
+  inherit (pkgs.lenovo-legion) version src;
+
+  hardeningDisable = [ "pic" ];
+
+  preConfigure = ''
+    sed -i -e '/depmod/d' ./kernel_module/Makefile
+  '';
+
+  makeFlags = kernel.makeFlags ++ [
+    "SHELL=${bash}/bin/bash"
+    "KERNELVERSION=${kernel.modDirVersion}"
+    "KSRC=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALLDIR=${placeholder "out"}/lib/modules/${kernel.modDirVersion}"
+    "MODDESTDIR=${placeholder "out"}/${kernel.modDirVersion}/kernel/drivers/platform/x86"
+    "DKMSDIR=${placeholder "out"}/lib/modules/${kernel.modDirVersion}/misc"
+  ];
+
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  preBuild = "cd kernel_module";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/platform/x86
+    install -p -D -m 444 *.ko $out/lib/modules/${kernel.modDirVersion}/kernel/drivers/platform/x86
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "A Linux kernel module for controlling fan and power in Lenovo Legion laptops";
+    homepage = "https://github.com/johnfanv2/LenovoLegionLinux";
+    license = lib.licenses.gpl2;
```

```suggestion
    license = lib.licenses.gpl2Only;
```

The non-specific GPL license attrs are deprecated.

---

Thank you